admin/class-admin-apple-json.php

```diff
@@ -61,17 +80,35 @@ public function __construct() {
 	 * @access public
 	 */
 	public function action_router() {
+
 		// Check for a valid action.
-		$action = isset( $_POST['apple_news_action'] ) ? sanitize_text_field( $_POST['apple_news_action'] ) : null;
+		$action = isset( $_REQUEST['apple_news_action'] )
+			? sanitize_text_field( wp_unslash( $_REQUEST['apple_news_action'] ) )
+			: null;
 		if ( ( empty( $action ) || ! array_key_exists( $action, $this->valid_actions ) ) ) {
 			return;
 		}
 
 		// Check the nonce.
 		check_admin_referer( 'apple_news_json' );
 
+		// Store the selected component value for use later.
+		$this->selected_component = isset( $_POST['apple_news_component'] )
+			? sanitize_text_field( wp_unslash( $_POST['apple_news_component'] ) )
+			: '';
+		if ( ! array_key_exists( $this->selected_component, $this->list_components() ) ) {
+			$this->selected_component = '';
+		}
+
+		// Store the selected theme for use later.
+		if ( ! empty( $_POST['apple_news_theme'] ) ) {
+			$this->selected_theme = sanitize_text_field( wp_unslash( $_POST['apple_news_theme'] ) );
+		}
+
 		// Call the callback for the action for further processing.
-		call_user_func( $this->valid_actions[ $action ]['callback'] );
+		if ( isset( $this->valid_actions[ $action ]['callback'] ) ) {
```

Comment:
Should you also check `is_callable( $this->valid_actions[ $action ]['callback'] )` here?